src/fonduer/learning/task.py

```diff
@@ -0,0 +1,192 @@
+import logging
+from functools import partial
+from typing import Any, Dict, List, Optional, Union
+
+import torch.nn as nn
+import torch.nn.functional as F
+from emmental.modules.embedding_module import EmbeddingModule
+from emmental.modules.rnn_module import RNN
+from emmental.modules.sparse_linear_module import SparseLinear
+from emmental.scorer import Scorer
+from emmental.task import EmmentalTask
+from torch import Tensor
+
+from fonduer.learning.modules.soft_cross_entropy_loss import SoftCrossEntropyLoss
+from fonduer.learning.modules.sum_module import Sum_module
+from fonduer.utils.config import get_config
+
+logger = logging.getLogger(__name__)
+
+
+sce_loss = SoftCrossEntropyLoss()
+
+
+def loss(
+    module_name: str,
+    intermediate_output_dict: Dict[str, Any],
+    Y: Tensor,
+    active: Tensor,
+) -> Tensor:
+    if len(Y.size()) == 1:
+        label = intermediate_output_dict[module_name][0].new_zeros(
+            intermediate_output_dict[module_name][0].size()
+        )
+        label.scatter_(1, (Y - 1).view(Y.size()[0], 1), 1.0)
+    else:
+        label = Y
+
+    return sce_loss(intermediate_output_dict[module_name][0][active], label[active])
+
+
+def output(module_name: str, intermediate_output_dict: Dict[str, Any]) -> Tensor:
+    return F.softmax(intermediate_output_dict[module_name][0])
+
+
+def create_task(
+    task_names: Union[str, List[str]],
+    n_arities: Union[int, List[int]],
+    n_features: int,
+    n_classes: Union[int, List[int]],
+    emb_layer: Optional[EmbeddingModule],
+    mode: str = "mtl",
+    model: str = "LSTM",
+) -> List[EmmentalTask]:
+    """Create task from relation(s).
+
+    :param task_names: Relation name(s), If str, only one relation; If List[str],
+        multiple relations.
+    :type task_names: str, List[str]
+    :param n_arities: The arity of each relation.
+    :type n_arities: int, List[int]
+    :param n_features: The multimodal feature set size.
+    :type n_features: int
+    :param n_classes: Number of classes for each task. (Only support classification
+        task now).
+    :type n_classes: int, List[int]
+    :param emb_layer: The embedding layer for LSTM. No need for LogisticRegression
+        model.
+    :type emb_layer: EmbeddingModule
+    :param mode: Model mode, defaults to "mtl".
```

Comment:
What are the options here? Just "mtl" and "stl"? Can we make it explicit so users don't have to guess?

Reply:
Actually, it seems this parameter isn't even used?